topostats/tracing/dnatracing.py

```diff
@@ -1512,31 +1542,47 @@ def analyse_nodes(self, box_length: float = 20e-9):
                     LOGGER.info(f"Node stats skipped as resolution too low: {self.px_2_nm}nm per pixel")
                     error = True
 
+                # image_slices = (single_branch_coords_img[:,0].min() - 10, single_branch_coords_img[:,0].max() + 10,
+                #                single_branch_coords_img[:,1].min() - 10, single_branch_coords_img[:,1].max() + 10)
+                image_slices = (
+                    x - int(max_length_px * 1.2),
+                    x + int(max_length_px * 1.2),
+                    y - int(max_length_px * 1.2),
+                    y + int(max_length_px * 1.2),
+                )
+
                 print("Error: ", error)
                 self.node_dict[real_node_count] = {
                     "error": error,
                     "px_2_nm": self.px_2_nm,
                     "crossing_type": None,
                     "branch_stats": matched_branches,
                     "node_stats": {
-                        "node_mid_coords": [x, y],
-                        "node_area_image": self.image[box_lims[0] : box_lims[1], box_lims[2] : box_lims[3]], # self.hess
-                        "node_area_grain": self.grain[box_lims[0] : box_lims[1], box_lims[2] : box_lims[3]],
-                        "node_area_skeleton": node_area,
-                        "node_branch_mask": branch_img,
-                        "node_avg_mask": avg_img,
+                        "node_coords": node_coords,
+                        "node_area_image": self.image[
+                            image_slices[0] : image_slices[1], image_slices[2] : image_slices[3]
+                        ],  # self.hess
+                        "node_area_grain": self.grain[
+                            image_slices[0] : image_slices[1], image_slices[2] : image_slices[3]
+                        ],
+                        "node_area_skeleton": reduced_node_area[
+                            image_slices[0] : image_slices[1], image_slices[2] : image_slices[3]
+                        ],
+                        "node_branch_mask": branch_img[
+                            image_slices[0] : image_slices[1], image_slices[2] : image_slices[3]
+                        ],
+                        "node_avg_mask": avg_img[image_slices[0] : image_slices[1], image_slices[2] : image_slices[3]],
```

Comment:
@MaxGamill-Sheffield This line produces an error in cases where `average_trace_advised` is false as `avg_img` is set to None in line 1511 and is not redefined after this point

Reply:
Ah great catch, thankyou Laura!